src/config.test.ts

```diff
@@ -4,46 +4,97 @@ import {
 } from '@jupiterone/integration-sdk-core';
 import {
   createMockExecutionContext,
-  setupRecording,
+  Recording,
 } from '@jupiterone/integration-sdk-testing';
 
 import { validateInvocation } from './config';
 import { IntegrationConfig } from './types';
+import { integrationConfig } from '../test/config';
+import { setupArtifactoryRecording } from '../test/helpers/recording';
 
-it('requires valid config', async () => {
-  const executionContext = createMockExecutionContext<IntegrationConfig>({
-    instanceConfig: {} as IntegrationConfig,
+describe('JFrog Arrifactory', () => {
+  let recording: Recording;
+
+  afterEach(async () => {
+    await recording.stop();
   });
 
-  try {
-    await validateInvocation(executionContext);
-  } catch (e) {
-    expect(e instanceof IntegrationValidationError).toBe(true);
-  }
-});
+  test('requires valid config', async () => {
+    recording = setupArtifactoryRecording({
+      directory: __dirname,
+      name: 'invalidConfig',
+      options: {
+        matchRequestsBy: {
+          url: {
+            hostname: false,
+          },
+        },
+        recordFailedRequests: false,
+      },
+    });
 
-it('auth error', async () => {
-  const recording = setupRecording({
-    directory: '__recordings__',
-    name: 'client-auth-error',
-  });
+    const executionContext = createMockExecutionContext<IntegrationConfig>({
+      instanceConfig: {} as IntegrationConfig,
+    });
 
-  recording.server.any().intercept((req, res) => {
-    res.status(401);
+    try {
+      await validateInvocation(executionContext);
+    } catch (e) {
+      expect(e instanceof IntegrationValidationError).toBe(true);
+    }
   });
 
-  const executionContext = createMockExecutionContext({
-    instanceConfig: {
-      clientNamespace: 'INVALID',
-      clientAccessToken: 'INVALID',
-      clientAdminName: 'INVALID',
-      enablePipelineIngestion: false,
-    },
+  test('auth error - 401', async () => {
+    recording = setupArtifactoryRecording({
+      directory: __dirname,
+      name: 'authError401',
+      options: {
+        matchRequestsBy: {
+          url: {
+            hostname: false,
+          },
+        },
+        recordFailedRequests: true,
+      },
+    });
+
+    const executionContext = createMockExecutionContext({
+      instanceConfig: integrationConfig,
+    });
+
+    executionContext.instance.config.clientAccessToken = 'badtoken';
+
+    try {
+      await validateInvocation(executionContext);
+    } catch (e) {
+      expect(e instanceof IntegrationProviderAuthenticationError).toBe(true);
+    }
```

Comment:
Testing async failures in jest can be tricky. In this case, if your call to `await validateInvocation(executionContext)` does _not_ fail, your test will still pass (it shouldn't).

Here's how async failures can be tested in jest:

```suggestion
    await expect(validateInvoation(executionContext).rejects.toThrow(IntegrationProviderAuthenticationError);
```

Reply:
Supah Smaht